compiler/rustc_error_messages/locales/en-US/infer.ftl

```diff
@@ -168,3 +168,104 @@ infer_msl_introduces_static = introduces a `'static` lifetime requirement
 infer_msl_unmet_req = because this has an unmet lifetime requirement
 infer_msl_trait_note = this has an implicit `'static` lifetime requirement
 infer_msl_trait_sugg = consider relaxing the implicit `'static` requirement
+
+infer_explicit_lifetime_required = explicit lifetime required in {$ident_kind ->
+    [ident] the type of `{$simple_ident}`
+    *[param_type] parameter type
+}
+    .label = lifetime `{$named}` required
+
+infer_explicit_lifetime_required_sugg = add explicit lifetime `{$named}` to {$ident_kind ->
+    [ident] the type of `{$simple_ident}`
+    *[param_type] type
+}
+
+infer_actual_impl_expl_1 = {$leading_ellipsis ->
+    [true] ...
+    *[false] {""}
+}{$kind ->
+    [signature] closure with signature `{$ty_or_sig}` must implement `{$trait_path}`
+    [passive] `{$trait_path}` would have to be implemented for the type `{$ty_or_sig}`
```

Comment:
I know that we've done this in previous pull requests, but it might make sense to try split these out into `enum` types and have the messages be distinct (I'd like to support `Diagnostic` derive on `enum`s to make this easier eventually).

Reply:
You mean turning `kind` strings into enums and implementing `IntoDiagnosticArg` for them?

Reply:
Not quite, turning `kind` strings into enums and implementing `IntoDiagnostic` for them, so that they are all their own distinct Fluent messages - I think it'll be easier for translators to translate that because otherwise they will need to check the source to know all of the different values something like "kind" could have currently.

Reply:
Oh, this would be great, but do we have a way to embed a diagnostic inside other diagnostic's message? Don't we need eager translation for that?

Reply:
I don't think we need to embed a diagnostic inside another, we now have `#[derive(Diagnostic)]` on enums (#102189) and we've had that on subdiagnostics from the start. I think we'd just want to try and enumerate all the different variations on a diagnostic in an enum - which is quite verbose, but  probably necessary for the translation.

Reply:
I've split `infer_actual_impl_expl_expected` and `infer_actual_impl_expl_but_actually` this way/ Should we also split anything else?